File: plugins/fedora/check_flex.rb
#!/usr/bin/env ruby
threshold = 0.90 #max ratio of licenses in use to licenses issued before WARNING

output = `/system/flexlm/lmstat -a`

matches = output.scan(/(Users of )([^:]*)(:  \(Total of )([0-9]*)( licenses issued; Total of )([0-9]*)( licenses in use\))/)

matches.each do |match|
	ratio = Float(match[5]) / float(match[3])
	if ratio == 1:
		puts "All " + match[1] + " licenses checked out"
		exit 2
	end
	if ratio > threshold:
		puts match[1] + " licenses are in high use"
		exit 1
	end
end
puts "Licenses are A-OK"
exit 0

File: check_flex.py
#!/usr/bin/python
import re,sys,commands

#Set Variables
threshold = 0.90 #The max ratio of licenses in use to licenses issued before WARNING

#Run lmstat
(status, output) = commands.getstatusoutput('/system/flexlm/lmstat -a')

#Create Pattern were looking for
#Regex Groups 
#Group 1: Name
#Group 3: Licenses Issued
#Group 5: Licenses in use
pattern = re.compile('(Users of )([^:]*)(:  \(Total of )([0-9]*)( licenses issued;  Total of )([0-9]*)( licenses in use\))')

#Perform the search and store matches in list
matches = re.findall(pattern, output)

#Parse each individual match for errors
for match in matches:
    ratio = float(match[5]) / float(match[3])
    if ratio == 1:
        print "All " + match[1] + " licenses checked out"
        sys.exit(2)
    if ratio > threshold:
        print match[1] + " licenses are in high use"
        sys.exit(1)


print "Licenses are A-OK"
sys.exit(0)
